test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -2442,10 +2440,4 @@ public Settings onNodeStopped(String nodeName) throws Exception {
             }
         });
     }
-
-    protected static void awaitDesiredBalanceShardsAllocator() throws Exception {
-        if (internalCluster().getCurrentMasterNodeInstance(ShardsAllocator.class)instanceof DesiredBalanceShardsAllocator dbsa) {
-            assertBusy(() -> assertTrue(dbsa.isIdle()));
```

Comment:
I think this is the only caller of `isIdle` so we can remove it and the flag on which it depends.

Reply:
Unfortunately it is still used in `org.elasticsearch.cluster.coordination.Coordinator#hasIdleJoinValidationService` and `org.elasticsearch.cluster.coordination.JoinValidationServiceTests#testConcurrentBehaviour`

Reply:
ah ok